File: MicroCli/src/main/resources/micronaut/security/api key/java/JDBC/domains/APIKey.java
package ${securityPackage}.domains;


import io.micronaut.data.annotation.*;
import io.micronaut.data.model.naming.NamingStrategies;

import java.time.Instant;

@MappedEntity(value = "keys", namingStrategy = NamingStrategies.Raw.class)
public class APIKey {

    @Id
     @GeneratedValue(GeneratedValue.Type.AUTO)
    private <% if(jdbc) out.print 'Long'%><% if(mongo) out.print 'String'%> id;
    private String name;
    private String secret;

    private Instant expiry;

    @DateCreated
    private Instant created;

    @DateUpdated
    private Instant updated;

    public APIKey(){}
    public APIKey(String name, String key){
        this.name = name;
        this.secret = key;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }



    public Long  getId() {
        return id;
    }

    public void setId(Long  id) {
        this.id = id;
    }

    public Instant getExpiry() {
        return expiry;
    }

    public void setExpiry(Instant expiry) {
        this.expiry = expiry;
    }

    public Instant getCreated() {
        return created;
    }

    public void setCreated(Instant created) {
        this.created = created;
    }

    public Instant getUpdated() {
        return updated;
    }

    public void setUpdated(Instant updated) {
        this.updated = updated;
    }

    public String getSecret() {
        return secret;
    }

    public void setSecret(String secret) {
        this.secret = secret;
    }
}

File: MicroCli/src/main/resources/micronaut/security/api key/kotlin/JDBC/domains/APIKey.kt
package ${securityPackage}.domains

import io.micronaut.data.annotation.*
import io.micronaut.data.model.naming.NamingStrategies
import java.time.Instant

@MappedEntity(value = "apikeys", namingStrategy = NamingStrategies.Raw::class)
class APIKey(var name: String, var key: String) {
    @Id
    @GeneratedValue(GeneratedValue.Type.AUTO)
    var id: Long? = null
    var expiry: Instant? = null

    @DateCreated
    var created: Instant? = null

    @DateUpdated
    var updated: Instant? = null

}